src/gfx_layout.cpp

```diff
@@ -259,7 +259,9 @@ Point Layouter::GetCharPosition(std::string_view::const_iterator ch) const
 
 		for (int i = 0; i < run.GetGlyphCount(); i++) {
 			/* Matching glyph? Return position. */
-			if ((size_t)charmap[i] == index) {
+			if ((size_t)charmap[i] >= index) {
+				/* If the index is passed then it's part of a ligature, so return the position of the combined glyph. */
+				if (i > 0 && (size_t)charmap[i] > index) i--;
```

Comment:
Might this need to be a loop when there are more than two character leading to a combined glyph/ligature? Like the ﬃ ligature.

Reply:
No, I don't think so. With a two character ligature, `charmap` looks like this: `[0, 1, 2, 4]`
We're looking for index 3, but got to 4 which is higher, so we go back one and get the position of 2.
With three characters it would look like: `[0, 1, 2, 5]`
We'd be looking for 3 (or 4), but get to 5. So again we go back one and get the position of 2.

Reply:
Hmm, it will fail if there's no character after the ligature though. I'll revise!